x-pack/plugins/profiling/server/routes/search_stacktraces.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+import { Logger } from '@kbn/logging';
+import { withProfilingSpan } from '../utils/with_profiling_span';
+import { DedotObject } from '../../common/elasticsearch';
+import {
+  Executable,
+  FileID,
+  StackFrame,
+  StackFrameID,
+  StackTrace,
+  StackTraceID,
+} from '../../common/profiling';
+import { ProjectTimeQuery } from './query';
+
+interface ProfilingEvents {
+  [key: string]: number;
+}
+
+type ProfilingStackTrace = DedotObject<{
+  ['file_ids']: string[];
+  ['frame_ids']: string[];
+  ['address_or_lines']: number[];
+  ['type_ids']: number[];
+}>;
+
+interface ProfilingStackTraces {
+  [key: string]: ProfilingStackTrace;
+}
+
+type ProfilingStackFrame = DedotObject<{
+  ['file_name']: string | undefined;
+  ['function_name']: string;
+  ['function_offset']: number | undefined;
+  ['line_number']: number | undefined;
+  ['source_type']: number | undefined;
+}>;
+
+interface ProfilingStackFrames {
+  [key: string]: ProfilingStackFrame;
+}
+
+interface ProfilingExecutables {
+  [key: string]: string;
+}
+
+type StackTraceResponse = DedotObject<{
+  ['stack_trace_events']: ProfilingEvents;
+  ['stack_traces']: ProfilingStackTraces;
+  ['stack_frames']: ProfilingStackFrames;
+  ['executables']: ProfilingExecutables;
+  ['total_frames']: number;
+}>;
+
+export async function searchStackTraces({
+  logger,
+  client,
+  filter,
+  sampleSize,
+}: {
+  logger: Logger;
+  client: ElasticsearchClient;
+  filter: ProjectTimeQuery;
+  sampleSize: number;
+}) {
+  return withProfilingSpan('search_stack_traces', async () => {
+    const response = await client.transport.request<StackTraceResponse>({
+      method: 'POST',
+      path: encodeURI('/_profiling/stacktraces'),
+      body: {
+        sample_size: sampleSize,
+        query: filter,
+      },
+    });
+
+    const stackTraceEvents: Map<StackTraceID, number> = new Map(
+      Object.entries(response.stack_trace_events)
```

Comment:
Immediately after startup I saw this:

```
[2022-12-07T10:16:39.301+01:00][ERROR][plugins.profiling] TypeError: Cannot convert undefined or null to object
    at Function.entries (<anonymous>)
    at ~/Projects/kibana/x-pack/plugins/profiling/server/routes/search_stacktraces.ts:83:14
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[...]
```

I assume that this happens when we return an empty response, i.e.:

```json
{
  "total_frames" : 0
}
```

As soon as stack trace events where trickling in, it was fine.

Reply:
Thanks for noticing that. I updated the code to handle missing fields. I also added unit tests to confirm this.